### PR TITLE
pkg: Go degraded if disk metrics are above a threshold  

### PR DESCRIPTION
**Pull request by lilic**
/hold

Still testing thresholds.

### CODE REVIEW

File: pkg/operator/metriccontroller/fsync_controller.go [hunk 1]
```diff
@@ -81,12 +87,18 @@ func (c *FSyncController) sync(ctx context.Context, syncCtx factory.SyncContext)
 		return fmt.Errorf("unexpected type, expected Matrix, got %T", matrix)
 	}
 
+	degraded := false
```
Comment (lilic) [inline pkg/operator/metriccontroller/fsync_controller.go:90]: Consider: degradedMsg string to append values to.

Comment (hexfusion) [reply pkg/operator/metriccontroller/fsync_controller.go:90]: + 1 yeah lets be very direct that we have observed performance thresholds which can endanger the stability of the cluster.

File: pkg/operator/metriccontroller/fsync_controller.go [hunk 2]
```diff
@@ -105,7 +117,37 @@ func (c *FSyncController) sync(ctx context.Context, syncCtx factory.SyncContext)
 	// Send warning event if excessive leader changes are detected. The event will include fsync disk metrics.
 	syncCtx.Recorder().Warningf("EtcdLeaderChangeMetrics", "Detected leader change increase of %s over 5 minutes on %q; disk metrics are: %s. Most often this is as a result of inadequate storage or sometimes due to networking issues.", leaderChanges, platformType, strings.Join(values, ","))
 
-	// TODO: Consider Degraded condition here.
+	// Capture any etcd alerts. We want to go degraded if critical etcd alerts are firing to warn the admin.
+	etcdAlertsResult, _, err := client.Query(ctx, "ALERTS{alertname=~\"etcd.+\", job=\"etcd\"}", time.Now())
```
Comment (hexfusion) [inline pkg/operator/metriccontroller/fsync_controller.go:121]: If there is a bug or corner case with alerts I think we risk upgrade disruption.

Comment (lilic) [reply pkg/operator/metriccontroller/fsync_controller.go:121]: Yes agreed, we can also skip going degraded on alerts for now until we stabilize the alerts. I can open a follow up jira ticket for the future releases? And for now only go degraded on leader changes and fsync?

Comment (hexfusion) [reply pkg/operator/metriccontroller/fsync_controller.go:121]: sgtm, thanks!

Comment (lilic) [reply pkg/operator/metriccontroller/fsync_controller.go]: Done, also added TODO in the code with the query that would be used. https://issues.redhat.com/browse/ETCD-239

**Comment by lilic** [comment]
/retest

Unrelated.
> fail [github.com/openshift/origin/test/extended/authorization/scc.go:73]: 3 pods failed before test on SCC errors
Error creating: pods "console-operator-59967bcc9-" is forbidden: unable to validate against any security context constraint: [provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount] for ReplicaSet.apps/v1/console-operator-59967bcc9 -n openshift-console-operator happened 14 times

**Pull request merged**